packages/core/test/input.test.ts

```diff
@@ -0,0 +1,101 @@
+import { interpret } from '../src';
+import { createTypes } from '../src/createTypes';
+import { createMachine2 as createMachine } from '../src/StateMachine';
+
+describe('machine.withInput()', () => {
+  it('should create a machine with input', () => {
+    expect.assertions(3);
+    const t = createTypes({
+      input: {} as { startCount: number },
+      context: {} as { count: number }
+    });
+
+    const machine = createMachine<typeof t>({
+      context: ({ input }) => ({
+        count: input.startCount
+      }),
+      entry: (ctx) => {
+        expect(ctx.count).toBe(42);
+      }
+    });
+
+    const inputMachine = machine.withInput({ startCount: 42 });
+    const initialState = inputMachine.getInitialState();
```

Comment:
I love this so much!

`withInput` is going to be a nice mechanism for cleaning up a lot of boilerplate, and defining types externally with `createTypes` is a little more clear than `schema` + I like that creating a generic machine is now just `typeof t` instead of needing to remember which params it accepts and in which order.